src/util.js

```diff
@@ -28,6 +31,25 @@ export function safeEmit(q, eventName, arg) {
   }
 }
 
+
+export function compressMessage(message, compression) {
+  if (compression === true || compression.encoding === supportedCompression) {
+    return {
+      headers: { 'Content-Encoding': supportedCompression },
+      body: zlib.deflateSync(JSON.stringify(message)),
```

Comment:
We should probably use the async call.

Also if we are sending text we might need to Base64 encode to send over SQS.  I am assuming what comes out of Zlib deflate is a byte array.